### ISSUE TITLE
Tensorrt model inference not working on folder of images and videos

### ISSUE THREAD

**Issue by vardanagarwal** [ultralytics/yolov5#9578]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Detection

### Bug

Running the default commands in the given [colab notebook](https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb). Successful in creating a Tensorrt model but unable to run detect.py.

![Screenshot from 2022-09-25 17-45-46](https://user-images.githubusercontent.com/35430842/192142892-690ece1b-20f3-4709-b272-001caa896add.png)

The same issue is faced on a folder of videos, however works on a single video.

### Minimal Reproducible Example

```
python3 detect.py --weights yolov5s.engine --imgsz 640 --device 0  # inference
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@vardanagarwal 👋 hi, thanks for letting us know about this possible problem with YOLOv5 🚀. I've run the code below and am able to reproduce the same issue you are seeing. I'll add a TODO and prioritize a fix for this.

**Comment by glenn-jocher** [comment]
@vardanagarwal good news 😃! Your original issue may now be fixed ✅ in PR #9581. I've also added exported model inference to our CI to catch this type of error earlier in the future.

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
